### ISSUE TITLE
Make sklearn adapter to use LabelBinarizer by default

### ISSUE THREAD

**Issue by kudkudak** [tensorflow/tensorflow#1981]
TensorFlowDNNClassifier doesn't handle cases, when target labels are not in `{0, 1}`. Usually `scikit-learn` models transform labels internally using `LabelBinarizer` so this is slightly misleading. 

``` python
from sklearn import datasets, metrics, cross_validation
from tensorflow.contrib import skflow

## Load dataset and select only first two classes
iris = datasets.load_iris()
ids = np.where((iris.target==0) | (iris.target==1))
iris.data = iris.data[ids]
iris.target = iris.target[ids]

## Change labels from 0/1 to -1/1
## Comment out the following line to get 1.0 accuracy
iris.target[iris.target==0] = -1 

## Train and predict on iris (copied from example)
X_train, X_test, y_train, y_test = cross_validation.train_test_split(iris.data, iris.target,
    test_size=0.2, random_state=42)

classifier = skflow.TensorFlowDNNClassifier(hidden_units=[10, 20, 10],
    n_classes=3, steps=200)

# Fit and predict.
classifier.fit(X_train, y_train)
score = metrics.accuracy_score(y_test, classifier.predict(X_test))
print('Accuracy: {0:f}'.format(score)) # Prints 0.4, but prints 1.0 without label change
```

TF version is `0.8.0rc0`

EDIT:

What happens is that `classifier.predict(X_test)` is in [1,2] instead of [-1, 1]

**Comment by suharshs** [comment]
@ilblackdragon Any update on this?

**Comment by ilblackdragon** [comment]
We can't do LabelEncoding for you due to support of streaming data - where we don't know what classes you have. Also when data is streaming through the graph estimators don't control the execution flow in anyway.

Because of that, please use encoding before the model using scikilearn.LabelEncoder or you can do it inside the model function, by using `tf.contrib.lookup.HashTable` lookups.

**Comment by kudkudak** [comment]
@ilblackdragon hmm. Some thoughts from outside point of view: standard support for cases like that in sklearn is passing classes as arguments (for instance in ``sklearn.metrics``). Why not add classes argument to fit and enforce it in the case of streaming data, and if data is not streaming, do the transformation? It seems slightly confusing for heavy sklearn users (as the example hopefully shows)?

**Comment by ilblackdragon** [comment]
@kudkudak We are actually planning to separate sklearn support into an adapter, where we can implement auto-relabelling magically indeed. I'll add this to the current working proposal. Thanks!

**Comment by drpngx** [comment]
OK, it sounds like the code might have moved around since this issue was last touched. Feel free to open a new github issue if the problem still persists in recent versions, referencing this one.
